src/edu/csus/ecs/pc2/imports/ccs/ICPCTSVLoader.java

```diff
@@ -319,17 +329,57 @@ public static Group[] loadGroups(String filename) throws Exception {
             } catch (NumberFormatException e) {
                 throw new InvalidValueException("Expecting group number got '" + numberStr + "' on line " + lines[i], e);
             }
-
-            Group group = new Group(groupName);
+            
+            Group group = findGroupById(authoritativeGroups, number);
+            if (group == null) {
+                group = new Group(groupName);
+            }
+            
+            group.setDisplayName(groupName);
             group.setGroupId(number);
-
-            groups[groupAccount] = group;
-            groupAccount++;
+            
+            groupList.add(group);
         }
-
+        
+        /**
+         * Add groups from authoritativeGroups if missing from groupList.
+         */
+        for (Group aGroup : authoritativeGroups) {
+            
+            Group missingGroup = null; 
+            
+            for (Group glGroup : groupList) {
+                if (aGroup.getGroupId() == glGroup.getGroupId()) {
+                    missingGroup = glGroup;
```

Comment:
`missingGroup` is a VERY bad/misleading name for this variable.  In fact, when a value is assigned to `missingGroup`, that  value is a Group which is NOT missing (from `groupList`).  The reversed logic of this is quite confusing (although I'll concede that the binary works...)

Reply:
Suggest a less misleading name, rather than just comment.   I used that name thinking at the time that
it made sense (or more likely the code section was rewritten keeping that name).

Reply:
Well, since what this loop is doing is checking, for each Group in `authoritativeGroups`, whether that group is found in `groupList` (and if NOT, it is adding the Group from `authoritativeGroups` into `groupList`, a "better" name might be "found".

However, the real problem is the confusion in what this loop is trying to accomplish vs. the code it uses to accomplish that.  What it is trying to accomplish (according to the comment at the top, and, unless I'm mistaken, in the code itself) is to add to `groupList` any Group which is currently in `authoritativeGroups` but is NOT already in `groupList`.  I think a better solution would be to replace the entire loop with something like:
```
       for (Group aGroup : authoritativeGroups) {
           if (! groupList.contains(aGroup) {
              groupList.add(aGroup);
              break;
          }
       }
```
This is much simpler and clearer, and I believe it accomplishes what is intended (although I would certainly expect you to review that assertion :)  )

Reply:
So no change in the name of the variable, ok.

Reply:
pushed fix.